app/controllers/grant_submissions/submissions_controller.rb

```diff
@@ -80,12 +97,34 @@ def set_grant
       @grant = Grant.friendly.find(params[:grant_id])
     end
 
+    def set_state(submission)
+      if saved_as_draft?
+        submission.state = 'draft'
+        # submission.update(state: 'draft')
+      elsif submitted?
+        submission.state = 'submitted'
+        submission.submitted_at = Time.now
+        # submission.update(state: 'submitted', submitted_at: Time.now)
+      else
+        flash[:error] = submission.errors.to_a
+        redirect_back fallback_location: grant_submissions_path(@grant)
+      end
```

Comment:
Could just use case statement on the param instead. Also, would prefer to use GrantSubmission::SUBMISSION_STATES keys instead of strings.